### ISSUE TITLE
[DOC-FIX] SFTP configuration example 

### ISSUE THREAD

**Issue by Sergiodiaz53** [mlflow/mlflow#5799]
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for information on what types of issues we address.

**Please fill in this documentation issue template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages documentation fix contributions. Would you or another member of your organization be willing to contribute a fix for this documentation issue to the MLflow code base?

- [ ] Yes. I can contribute a documentation fix independently.
- [X ] Yes. I would be willing to contribute a document fix with guidance from the MLflow community.
- [ ] No. I cannot contribute a documentation fix at this time.

### URL(s) with the issue:

https://www.mlflow.org/docs/latest/tracking.html

### Description of proposal (what needs changing):
The documentation of MLFlow regarding artifact store is pretty basic.
In my specific case, I'm trying to setup it in using SFTP but there some specific things need to be mentioned.

Basically I just want to connect to an SFTP running in x.x.x.x:9222 with a specific private key. Is that even possible in the current implementation?

There is no explanation on how to use specific IdentityFiles, I have already try to define then in ~/.ssh/config with no luck but, after checking the source code, I saw that it will use a the default file called 'id_rsa' if none it is specified, so It did the trick.Regarding the custom port issue, I havent found a solution to solve it yet.

I will be glad to contribute with a working example for future users with MLFlow + SFTP if I achieve this but I will need some help.

Thanks in advance.

**Comment by dbczumar** [comment]
Hi @Sergiodiaz53, using `~/.ssh/config` is indeed the recommended approach for specifying custom private keys and other host-specific connection properties. To confirm, is the hostname included in the artifact URI? If you don't mind doing some printline debugging, perhaps printing the value of `user_config` in https://github.com/mlflow/mlflow/blob/d0d307f7f7b49f013727191a672ae2139bf37343/mlflow/store/artifact/sftp_artifact_repo.py#L52 could shed some light on the situation. Once this is resolved, we would be very excited about a documentation improvement; thank you so much in advance.

**Comment by Sergiodiaz53** [comment]
Hi dbczumar, 

Sorry for the delay.
I'm running MLFLOW in a docker with a exposed port to localhost in :5000
I'm running SFTP in a remote host 2.2.2.2:9222
I'm running my CLIENT in localhost

Im launching MLFlow inside my docker container as root using: 
`mlflow server  --port 5000  --host 0.0.0.0  --backend-store-uri sqlite:///db/test}.db --default-artifact-root sftp://mlflow@ 2.2.2.2:9222`

I'm launching my CLIENT using my ubuntuuser, inside myfile.py I set up:

```
remote_server_uri = "http://127.0.0.1:5000"
mlflow.set_tracking_uri(remote_server_uri)

[train model]

mlflow.keras.log_model(...)
```

I have added to ~/.ssh/config in the docker container with mlflow and my localhost this information:

```
Host sftp_server
        Hostname 2.2.2.2
        User mlflow
        IdentityFile /c/Users/myuser/.ssh/id_rsa
```

When I run myfile.py, the print you told me (inside the sftp_artifact_repo) is priting:
{ ''hostname'' : ''2.2.2.2''}
and being id_rsa my only rsa I have seen that is being taken.
So what is probable happening is that Im trying to connect to 2.2.2.2 (OK) with the user mlflow (OK) and the id_rsa file (OK) but in the port 22 and NOT the 9222.

Questions:
**1)** Is there anyway to setup a custom port for SFTP? Do I add the port into the ~/.ssh/config file?

```
Host sftp_server
        Hostname 2.2.2.2
        Port 9222
        User mlflow
        IdentityFile /c/Users/myuser/.ssh/id_rsa
```

**2)** Knowing that Im running mlflow inside a docker container, I think I should set the ~/.ssh/config file in both (meaning the localhost and the docker itself), am I right?

Thanks in advance!

**Comment by Sergiodiaz53** [comment]
Hi there,

I had some advances on this.
At first let me say that if I run:

`sftp mlflow@2.2.2.2`

everything works and I can create files and directories in my sftp server. So without MLFlow everything seems to work fine (meaning that my .ssh/config file is ok)

Now when I print my user_config inside sftp_artifact_repo.py what I get is:

{'hostname': '2.2.2.2', 'port': '9222', 'user': 'mlflower', 'identityfile': ['/home/ubuntu/.ssh/id_rsa_sftp']}

So I think I'm on the good way.
The problem is I'm getting a:

`paramiko.ssh_exception.SSHException: Bad host key from server`

Due to the user_config print I'm pretty sure we are using the correct key, may the problem be related to the known_hosts?

Im running my .py using my user UBUNTUUSER but I'm connecting using the user MLFLOW, I have already try to create a /home/mlflow/.ssh/known_hosts with the fingerprint but I still have the same problem.

Hope you can help! 
Thanks in advance
